metaflow/plugins/kfp/kfp.py

```diff
@@ -0,0 +1,113 @@
+import kfp
+from kfp import dsl
+
+from .constants import DEFAULT_RUN_NAME, DEFAULT_EXPERIMENT_NAME, DEFAULT_FLOW_CODE_URL, DEFAULT_KFP_YAML_OUTPUT_PATH
+
+def get_ordered_steps(graph):
+    """
+    Returns the ordered step names in the graph (FlowGraph) from start step to end step as a list of strings containing the
+    step names.
+
+    # TODO: Support other Metaflow graphs, as branching and joins are not currently supported
+    Note: All MF graphs start at the "start" step and end with the "end" step (both of which are mandatory).
+    """
+
+    ordered_steps = ['start']
+    current_node_name = 'start'
+
+    # This is not an ideal way to iterate over the graph, but it's the (easy+)right thing to do for now.
+    # This may need to change as work on improvements.
+    while current_node_name != 'end':
+        for node in graph:
+            if node.name == current_node_name:
+                if current_node_name != 'end':
+                    current_node_name = node.out_funcs[0]
+                    ordered_steps.append(current_node_name)
+                    break
+
+    return ordered_steps
+
+
+def step_container_op(step_name, code_url=DEFAULT_FLOW_CODE_URL):
+    """
+    Method to create a kfp container op to run a single step (here, we also execute our custom pre-start step
+    for setup as we aren't maintaining state) and then execute the given step.
+
+    TODO: This does not maintain state. The custom pre-start command used below would be removed once we have state accessible across KFP steps.
+    TODO: The public docker is a copy of the internal docker image we were using (borrowed from aip-kfp-example). Check if any stage here may need to be further modified later.
+    """
+
+    python_cmd = """ "python helloworld.py --datastore-root ", $1, " step {} --run-id ", $2, " --task-id ", $4, " --input-paths", $2"/"$5"/"$6 """.format(
+        step_name)
+    command_inside_awk = """ {{ print {0} }}""".format(python_cmd)
+    final_run_cmd = """ python helloworld.py pre-start | awk 'END{}' | sh """.format(command_inside_awk)
+
+    return dsl.ContainerOp(
+
+        name='StepRunner-{}'.format(step_name),
+        image='ssreejith3/mf_on_kfp:python-curl-git',
+        command= ['sh', '-c'],
+        arguments=[
+            'curl -o helloworld.py {}' \
+            ' && pip install git+https://github.com/zillow/metaflow.git@c722fceffa3011ecab68ce319cff98107cc49532' \
+            ' && export USERNAME="kfp-user" ' \
+            ' && {}'.format(code_url, final_run_cmd)
```

Comment:
Anything stopping us from using f-strings instead of `format()`? As far as I remember f-strings are available in Python 3.0+ so hopefully we don't need to support 2.7 anymore 😅